meilisearch/index.py

```diff
@@ -1577,6 +1577,66 @@ def reset_faceting_settings(self) -> TaskInfo:
 
         return TaskInfo(**task)
 
+    # USER DICTIONARY SUB-ROUTES
+
+    def get_dictionary(self) -> List[str]:
+        """Get the dictionary entries of the index.
+
+        Returns
+        -------
+        settings:
+            List containing the dictionary entries of the index.
+
+        Raises
+        ------
+        MeilisearchApiError
+            An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://www.meilisearch.com/docs/reference/errors/error_codes#meilisearch-errors
+        """
+        return self.http.get(self.__settings_url_for(self.config.paths.dictionary))
+
+    def update_dictionary(self, body: Union[List[str], None]) -> TaskInfo:
```

Comment:
Body is required here.

```suggestion
    def update_dictionary(self, body: List[str]) -> TaskInfo:
```

Reply:
Alright, I pushed this change. 
Is there some reference that could have allowed me to prevent this? From what I could see most other List settings allowed None, I'm not too experienced with Python so I tried not to do my own thing.

Reply:
If it is something that has been released like this you can find it in the API reference. In this case [here](https://www.meilisearch.com/docs/reference/api/settings#update-dictionary). If it could be `None` there will be a default value, something like [this](https://www.meilisearch.com/docs/reference/api/search#body).

If it is pre-release they create an issue in the `integration-guide` repo where you can find information about the needed changes. Something like [this](https://github.com/meilisearch/integration-guides/issues/290).

Reply:
I apologize for going on about this, but I wanted to get it straightened out before writing the text separator subroutes.
 
This (null update) request actually successfully clears the dictionary for a 'books' index:
```bash
curl \
  -X PUT 'http://localhost:7700/indexes/books/settings/dictionary' \ 
  -H "Content-Type: application/json" --data-binary "null"
  
{"taskUid":66184,"indexUid":"books","status":"enqueued","type":"settingsUpdate","enqueuedAt":"2023-10-24T12:09:45.811281464Z"}
```

That request is what would be sent since
```python
# _http_requests.py > send_request
data=json.dumps(body) if body else "" if body == "" else "null",
```
represents `None` as `null` in the payload. Dictionary does in fact have a default value in the API documentation:

> [Reset dictionary](https://www.meilisearch.com/docs/reference/api/settings#reset-dictionary)
> `DELETE/indexes/{index_uid}/settings/dictionary`
> 
> Reset an index's dictionary to its **default value, [].**

Reply:
Fair enough, I guess technically it works but it wouldn't 100% do what the user expects. `[]` and `null` aren't the same thing, so if they want to remove the values they have added they should really be using the `reset` route.

Reply:
I agree, `reset` properly communicates its intentions (or `update` with `[]` for that matter), besides a variable being unexpectedly null is a very common programming mistake which would result in clearing this whole setting.

Should we create an issue to also remove the `None` option for other similar settings?

Reply:
In a lot of cases the `None` does make sense. Taking the `Index` `create` method as an example, `options: Optional[Dict[str, Any]] = None` makes sense because `None` in this case means not to attach any addtional `options`. In other words, some routes can take additional parameters but don't have to. So `None` in this case means there are no additional `options` to attach to the request.

In this case, even if `null` gets set to the server, the user is saying they want to take the defaults by not attaching additional parameters so they get what they expect.

Reply:
I suppose the distinction would be whether or not it makes sense to call the function without that argument. `Index` `create` can reasonably be called without `options` and use its defaults, however:
```python
    def update_searchable_attributes(self, body: Union[List[str], None]) -> TaskInfo:
```
calling this method with a body of `None` will set searchable_attributes to all (`[*]`).

Reply:
Your distinction is correct to me...but I found some context around why they are set like they are. https://github.com/meilisearch/meilisearch-python/issues/762#issuecomment-1559304152.

So while I think the way we have gone about this PR makes the most sense to me, I suppose we should allow the `None` option as you originally had it to be consistant with what the team wants. Sorry about that :frowning_face: